src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -13,14 +14,16 @@
 
 namespace System.Net.Http
 {
-    internal sealed partial class Http2Connection 
+    internal sealed partial class Http2Connection
     {
         private sealed class Http2Stream : IValueTaskSource, IDisposable
         {
             private enum StreamState : byte
             {
                 ExpectingHeaders,
                 ExpectingData,
+                ExpectingMoreData,
```

Comment:
What's the benefit of having ExpectingMoreData? Shouldn't the status just stay at ExpectingData through multiple data frames?

Reply:
There were test checking that for example header frame cannot be received after headers complete flag. Now it is ok if you get at least one data frame.

Reply:
Is that test necessary? A response that has headers and trailing headers but no content is (as far as I can tell) still valid. It's a weird situation, but it might be something we run into with gRPC where the trailers are used for status and metadata.

Here's the closes thing I could find to a relevant section of the spec:
```
An HTTP message (request or response) consists of:

   1.  for a response only, zero or more HEADERS frames (each followed
       by zero or more CONTINUATION frames) containing the message
       headers of informational (1xx) HTTP responses (see [RFC7230],
       Section 3.2 and [RFC7231], Section 6.2),

   2.  one HEADERS frame (followed by zero or more CONTINUATION frames)
       containing the message headers (see [RFC7230], Section 3.2),

   3.  zero or more DATA frames containing the payload body (see
       [RFC7230], Section 3.3), and

   4.  optionally, one HEADERS frame, followed by zero or more
       CONTINUATION frames containing the trailer-part, if present (see
       [RFC7230], Section 4.1.2).
```

Reply:
I'll take a look. Without it, test @geoffkizer added recently would fail. 
One option would be to track state with names same as RFC.
I'm not 100% is "ExpectingData represents Open state. 100%.

Reply:
After looking at the code, the tests I would expect to fail are `ResponseStreamFrames_HeadersAfterHeadersWithEndHeaders_ConnectionError` and `ResponseStreamFrames_HeadersAfterHeadersAndContinuationWithEndHeaders_ConnectionError`.

The tests seem fairly reasonable, but I don't think they have a basis in the spec.